### ISSUE TITLE
Is it worth raising issues / PRs?

### ISSUE THREAD

**Issue by ColinEberhardt** [AssemblyScript/assemblyscript#117]
I noticed your are working on v.Next of this compiler. I wondered whether it is worth raising issues or PRs against this version? Or whether it will become obsolete soon?

I've been playing around with AssemblyScript, and think it is a great idea. I have a few projects I've been working on - one is a re-implementation of d3 force layout:

https://github.com/ColinEberhardt/d3-wasm-force

I've spotted a number of bugs which I would be happy to raise sa issue, or try to fix.

**Comment by dcodeIO** [comment]
That's right, I am currently focusing on NEXT. I can't guarantee that I'll find a lot of time for review but when a PR is fine as is, it shouldn't be a problem to hit merge.

**Comment by dcodeIO** [comment]
Btw., for what it's worth, when looking through your code, I found `// TODO: cannot cast integers to floats, the required conversion functions are not exposed`.

That could, theoretically, be emulated in a portable way by using a Float64Array(1), like so:

```ts
const conversionBuffer: Float64Array = new Float64Array(1);

function convert(v: i32): f64 {
  return conversionBuffer[0] = v as f64;
}
```

When run as JS, the buffer will do the conversion (the `as` does nothing). When running in WASM, the `as` conversion should convert through the respective opcode (the buffer assignment is redundant).

**Comment by ColinEberhardt** [comment]
Thanks for the tip! Just tried it out - you cannot return the assignment, but otherwise, works a charm!

~~~javascript
function convert(v: i32): f64 {
  const conversionBuffer: Float64Array = new Float64Array(1);
  conversionBuffer[0] = v as f64;
  return conversionBuffer[0];
}
~~~
